packages/vite/src/node/ssr/ssrExternal.ts

```diff
@@ -179,8 +179,25 @@ function collectExternals(
 
 export function shouldExternalizeForSSR(
   id: string,
-  externals: string[]
+  externals: string[],
+  noExternals: string | RegExp | (string | RegExp)[] | true
 ): boolean {
+  const noExternalsArr: RegExp[] =
+    noExternals !== true
+      ? (Array.isArray(noExternals) ? noExternals : [noExternals]).map(
+          (item) => {
+            if (typeof item === 'string') {
+              item = new RegExp(item)
```

Comment:
I don't think you can just convert strings to regexes like this. The strings are meant to be exact matches. If the user wants to provide a regex they can simply pass that to begin with

Reply:
> I don't think you can just convert strings to regexes like this. The strings are meant to be exact matches. If the user wants to provide a regex they can simply pass that to begin with

Has updated